### PR TITLE
Qa/dynamic image component

### PR DESCRIPTION
**Pull request by robert-brice**
**Why are these changes introduced?**
* Related story #([URL](https://alm-tfs.apps.ci.gc.ca/tfs/IRCC/Scrum/_workitems/edit/939581))
* QA Link #(https://d1whfh0luwluq8.cloudfront.net/qa-dynamic-image-component/index.html)

**What is this pull request doing?**
* Creates a new component to handle images dynamically
* Provide the desired breakpoint and image path and the images will change at each viewport width
* Updates the icon button page to use new component on the 3 images flagged by design
* Enables native lazy loading of images optionally
* Fixes icon button design bug adding in figma link text
* Updates min/max width, anatomy and spec section images where necessary for input, date picker, spinner, banner, and select.

TESTING
* Navigate to the Icon Button Documentation page
* Open the inspector and network tab
* If you slowly scroll down the page, you'll see 3 images fetched in the network tab - lazy loaded
* The 3 images have a mobile breakpoint below 768px, where you will see the image change from the horizontal image to a vertical image provided by design

**Reviewer checklist:**
* Module structure is appropriate (when applicable)
* @Input()s are handled in keeping with established norms (i.e. a config and individual inputs for CL components)

** File names and hierarchies are in keeping with our norms:
* Interfaces start with 'I' followed by a capital letter and camel case (IInterfaceExample) and are descriptive
* Subjects are suffixed with 'Subj' (ex: thisIsAnExampleSubj)
* Subscriptions are suffixed with 'Sub' (ex: thisIsAnExampleSub)
* Observables ere suffixed with 'Obs$' (ex: thisIsAnExampleObs$)
* Class and variable names are camel case
* Class names should start with a capital letter (i.e. ExampleClass)
* Variable names should start with a lower case letter (i.e thisIsAnExampleVariable)
* All caps and underscores are used for exported constants (THIS_IS_A_CONSTANT)
* Names are all spelled correctly
* ngOnInit and constructor is removed when not used
* ':void' is removed from void functions
* Ensure the code contain appropriate media queries and accessibility elements
* Is the acceptance criteria met?

**Review component stylesheets:**
* New sheet is added to index.scss.
* New sheet includes @create and @selector(s) mixins.
* Selectors: Are classes leveraged as much as possible? Does the naming convention make sense and follow some sort of convention? (Nice to have: BEM naming applied)
* Is any styling repeated in the sheet? If yes can a mixin be leveraged instead?
* No !important tags are present in the page.
* All colors used are existing tokens. No mix-token mixin is leveraged unless creating a new token.
* Are the styles escaped using the template-const file.
* Avoid using element selectors for specificity where possible

**Token sheets:**
* All tokens are created using mix-token mixin.
* No layout styling is handled.
* Tokens are available globally at the project root.

### CODE REVIEW

**Review by MikeJalow — APPROVED** [review]

File: component-library/src/app/components/responsive-image/responsive-image.component.ts
Comment (kristopher-houston) [inline component-library/src/app/components/responsive-image/responsive-image.component.ts:1]: I thought we were putting this in the Component Library?

File: component-library/src/app/components/responsive-image/responsive-image.component.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  src: string;
+}
+@Component({
+  selector: 'app-responsive-image',
+  templateUrl: './responsive-image.component.html',
+  styleUrls: ['./responsive-image.component.scss']
+})
+export class ResponsiveImageComponent implements OnInit, AfterViewInit {
+  @ViewChild('image', { static: true })
+  image!: ElementRef<HTMLImageElement>;
```
[the remark this answers is not in the document]
Comment (kristopher-houston) [reply component-library/src/app/components/responsive-image/responsive-image.component.ts:31]: Or, set a value

File: component-library/src/app/components/responsive-image/responsive-image.component.ts
Comment (robert-brice) [reply component-library/src/app/components/responsive-image/responsive-image.component.ts:1]: I can certainly move it over, I must have misunderstood and thought it was just going into docsite for the time being.

**Comment by alexandreGrenierIRCC** [comment]
The QA link does not work for this PR, it redirects to the homepage.

**Pull request merged**